src/main/java/com/spotify/docker/client/DockerUtils.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.docker.client;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.nio.file.Paths;
+import java.util.Locale;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.System.getProperty;
+import static java.lang.System.getenv;
+
+/**
+ * A class that contains Docker constants and utility functions.
+ */
+class DockerUtils {
+
+  private SystemDelegate systemDelegate;
```

Comment:
if you set this to a default implementation then you don't have to have null checks below

```java
private SystemDelegate systemDelegate = new SystemDelegate() {
  @Override
  String getProperty(String key) {
    return System.getProperty(key);
  }
  @Override
  String getenv(String name) {
    return System.getenv(name);
  }
};
```